R/derive_vars_dy.R

```diff
@@ -0,0 +1,89 @@
+#' Derive Analysis Relative Day
+#'
+#' Adds the analysis relative day (`--DY`) to the dataset, i.e., study
+#' day of analysis date.
+#'
+#' @param dataset Input dataset
+#'
+#'   The columns specified by the `reference_date` and the `source_vars` parameter are
+#'   expected.
+#'
+#' @param reference_date The start date column, e.g., date of first treatment
+#'
+#'   A date or date-time object column is expected.
+#'
+#' @param source_vars A list of datetime or date variables created using `vars()` from
+#'   which dates are to be extracted. This can either be a list of date(time) variables
+#'   or named `--DY` variables and corresponding --DT(M) variables e.g.
+#'    vars(TRTSDTM, ASTDTM, AENDT) or vars(TRTSDT, ASTDTM, AENDT,DEATHDY=DTHDT)
+#'
+#' @author Teckla Akinyi
+#'
+#' @details The study day is derived as number of days from the reference date
+#'   to the end date. If it is nonnegative, one is added. I.e., the study day of the
+#'   reference date is 1. The input ---DT(M) is converted to ---DY
+#'
+#' @return The input dataset with `--DY` corresponding to the `--DTM` or `--DT`
+#'     source variable(s) added
+#'
+#' @keywords derivation ADaM timing
+#'
+#' @export
+#'
+#' @examples
+#' library(lubridate)
+#' library(dplyr)
+#'
+#' datain <- tibble::tribble(
+#'  ~STUDYID, ~USUBJID, ~TRTSDTM, ~ASTDTM, ~AENDT,
+#'  "TEST01", "PAT01", "2014-01-17T23:59:59", "2014-01-18T13:09:O9", "2014-01-20"
+#' ) %>%
+#'  mutate(TRTSDTM = lubridate::as_datetime(TRTSDTM),
+#'         ASTDTM = lubridate::as_datetime(ASTDTM),
+#'         AENDT = lubridate::ymd(AENDT))
+#'
+#' derive_vars_dy(datain, reference_date = TRTSDTM, source_vars = vars(TRTSDTM, ASTDTM, AENDT))
```

Comment:
Unfortunately, the new variables `ASTDY` and `AENDY` are not displayed. Maybe removing `STUDYID` and `USUBJID` (or moving them to the end) helps.

Reply:
agreed